utils.c

```diff
@@ -5,10 +5,46 @@
 #include <unistd.h>
 #include <sys/stat.h>
 #include <sys/ioctl.h>
-#include <linux/fs.h>
 
 #include "utils.h"
 
+
+#ifdef USE_LINUX_FS
```

Comment:
Detecting this via `CHECK_INCLUDE_FILE` feels a little fragile to me. I'd rather distinguish between the different OSes explicitly.

See for example https://github.com/samangh/gptfdisk/blob/master/diskio-unix.cc#L445-L479

Reply:
It felt like the best way to me, but I guess it's a question of taste. I changed it to just check for `__APPLE__`, which also removes quite a bit of cmake sourcery 👍